### ISSUE TITLE
Ionic setup fails out of the box / Compile error: Cannot redeclare block-scoped variable __DEBUG_BUILD__

### ISSUE THREAD

**Issue by zarko-tg** [getsentry/sentry-capacitor#394]
### Environment

Following the instructions from https://docs.sentry.io/platforms/javascript/guides/capacitor/ionic/

Node 18.16.1
macOS Ventura 13.4.1
Installed global @ionic/cli@7.1.1

### Steps to Reproduce

1. Created a fresh Ionic 7 / Angular 16 / Capacitor 5 project using "ionic start".
2. Followed the instructions from the docs link above.
3. ionic serve

Yields:
```
[ng] Error: node_modules/@sentry/capacitor/node_modules/@sentry/types/types/globals.d.ts:2:11 - error TS2451: Cannot redeclare block-scoped variable '__DEBUG_BUILD__'.
[ng]
[ng] 2     const __DEBUG_BUILD__: boolean;
[ng]             ~~~~~~~~~~~~~~~
[ng]
[ng]   node_modules/@sentry/types/types/globals.d.ts:2:11
[ng]     2     const __DEBUG_BUILD__: boolean;
[ng]                 ~~~~~~~~~~~~~~~
[ng]     '__DEBUG_BUILD__' was also declared here.
[ng]
[ng]
[ng] Error: node_modules/@sentry/types/types/globals.d.ts:2:11 - error TS2451: Cannot redeclare block-scoped variable '__DEBUG_BUILD__'.
[ng]
[ng] 2     const __DEBUG_BUILD__: boolean;
[ng]             ~~~~~~~~~~~~~~~
[ng]
[ng]   node_modules/@sentry/capacitor/node_modules/@sentry/types/types/globals.d.ts:2:11
[ng]     2     const __DEBUG_BUILD__: boolean;
[ng]                 ~~~~~~~~~~~~~~~
[ng]     '__DEBUG_BUILD__' was also declared here.
```

[sentest.zip](https://github.com/getsentry/sentry-capacitor/files/11833545/sentest.zip)

Attached you can find a reproduction project, where after unpacking you just run: npm i && ionic serve

**Comment by lforst** [comment]
This needs to be fixed upstream. https://github.com/getsentry/sentry-capacitor/issues/249

**Comment by g-apparence** [comment]
Adding "@sentry/types": "7.56.0" to the package.json fixed it for me.

**Comment by lucas-zimerman** [comment]
You need to alter your package.json
```
    "@sentry/angular-ivy": "^7.56.0",
    "@sentry/capacitor": "^0.12.1",
```
to
```
    "@sentry/angular-ivy": "7.56.0",
    "@sentry/capacitor": "0.12.1",
```
Sentry Capacitor has conflicts if a sibling package reference has a caret symbol.

**Comment by zVynx** [comment]
I cant fix the issue my package.json looks like this:

    "@sentry/angular-ivy": "7.56.0",
    "@sentry/capacitor": "0.12.1",
    "@sentry/types": "7.56.0",

i am using ionic 7.3.2 and capacitore core ^5.0.0 has anyone found a fix for this Bug?

**Comment by zVynx** [comment]
Nvm I got it working using

  "@sentry/angular-ivy": "7.54.0",
  "@sentry/capacitor": "0.12.1",

**Comment by BillBu** [comment]
Previously working but now reintroduced with the following in package.json:
    "@sentry/angular-ivy": "^7.71.0",
    "@sentry/capacitor": "^0.13.0-beta.1",

**Comment by Sampath-Lokuge** [comment]
@BillBu Until they'll fix this you can try this:

```
"@sentry/angular-ivy": "7.54.0",
"@sentry/capacitor": "0.12.1",
```

**Comment by BillBu** [comment]
> @BillBu Until they'll fix this you can try this:
> 
> ```
> "@sentry/angular-ivy": "7.54.0",
> "@sentry/capacitor": "0.12.1",
> ```

Thanks, but we need to use the 0.13 version because it fixes the issue where iOS is initialized on a background thread, which causes our app to crash. So we have to use that version (or at least get to the point where we can test it to see if it works).

I've uninstalled @sentry/angular-ivy and just referencing 

import * as SentryAngular from '@sentry/angular';

instead of 

import * as SentryAngular from '@sentry/angular-ivy';

And that compiles okay. Not sure what the effect of that is though and if that causes any other problems.

**Comment by Sampath-Lokuge** [comment]
@BillBu 

>import * as SentryAngular from '@sentry/angular';

The problem is this:

> Important: This package is not compatible with Angular 16 or newer. Please use [@sentry/angular-ivy](https://github.com/getsentry/sentry-javascript/tree/master/packages/angular-ivy) instead.

Ref: https://www.npmjs.com/package/@sentry/angular#angular-version-compatibility

**Comment by BillBu** [comment]
Thanks, having to revert to 0.12 until this __DEBUG_BUILD__ issue is fixed for 0.13.

**Comment by lucas-zimerman** [comment]
If you still are having this issue on the next release please let us know, for the time being I am closing this issue, thanks for opening it!
